File: DecompiledGameScripts/core/gameplay/prereqs/items/PlayerDoesntHaveRecipePrereq.swift

public class PlayerDoesntHaveRecipePrereq extends IScriptablePrereq {

  public let m_recipeID: TweakDBID;

  public let m_invert: Bool;

  protected func Initialize(recordID: TweakDBID) -> Void {
    this.m_recipeID = TDBID.Create(TweakDBInterface.GetString(recordID + t".recipe", ""));
    this.m_invert = TweakDBInterface.GetBool(recordID + t".invert", false);
  }

  protected const func OnApplied(state: ref<PrereqState>, game: GameInstance, context: ref<IScriptable>) -> Void {
    let craftingSystem: ref<CraftingSystem> = CraftingSystem.GetInstance(game);
    let playerCraftBook: ref<CraftBook> = craftingSystem.GetPlayerCraftBook();
    let result: Bool = !craftingSystem.IsRecipeKnown(this.m_recipeID, playerCraftBook);
    (state as PlayerDoesntHaveRecipePrereqState).OnChanged(result);
  }

  public const func IsFulfilled(game: GameInstance, context: ref<IScriptable>) -> Bool {
    let craftingSystem: ref<CraftingSystem> = CraftingSystem.GetInstance(game);
    let playerCraftBook: ref<CraftBook> = craftingSystem.GetPlayerCraftBook();
    let result: Bool = !craftingSystem.IsRecipeKnown(this.m_recipeID, playerCraftBook);
    if this.m_invert {
      return !result;
    };
    return result;
  }
}

File: Definitions/Game_Definitions/classes/PlayerDoesntHaveRecipePrereq.lua
---@meta
---@diagnostic disable

---@class PlayerDoesntHaveRecipePrereq : gameIScriptablePrereq
---@field recipeID TweakDBID
---@field invert Bool
PlayerDoesntHaveRecipePrereq = {}

---@return PlayerDoesntHaveRecipePrereq
function PlayerDoesntHaveRecipePrereq.new() return end

---@param props table
---@return PlayerDoesntHaveRecipePrereq
function PlayerDoesntHaveRecipePrereq.new(props) return end

---@param recordID TweakDBID|string
function PlayerDoesntHaveRecipePrereq:Initialize(recordID) return end

---@param context IScriptable
---@return Bool
function PlayerDoesntHaveRecipePrereq:IsFulfilled(context) return end

---@param state gamePrereqState
---@param context IScriptable
function PlayerDoesntHaveRecipePrereq:OnApplied(state, context) return end

